server/store.js

```diff
@@ -45,8 +45,19 @@ var store = _.extend(new events.EventEmitter(), {
       console.log('adding ' + infoHash);
 
       var e = engine(torrent, options);
+      var onready = function () {
+        var index;
+        if (typeof index !== 'number') {
```

Comment:
good point. I copied blindly from peerflix. I have simplified this code in the additional commit.